incompatibility/incompatibility-report/chain.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2021 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package incompatibility
+
+import (
+	openapiv3 "github.com/googleapis/gnostic/openapiv3"
+)
+
+var IncompatibilityChains []ChainIncompatibilitySearch = []ChainIncompatibilitySearch{
+	DocumentBaseSearch,
+	PathsSearch,
+}
+
+type ChainIncompatibilitySearch func(*openapiv3.Document) (incompatibilities []*Incompatibility)
+
+func SearchChains(doc *openapiv3.Document, chains ...ChainIncompatibilitySearch) *IncompatibilityReport {
+	var incompatibilities []*Incompatibility
+	for _, chain := range chains {
+		incompatibilities = append(incompatibilities, chain(doc)...)
+	}
+	return &IncompatibilityReport{Incompatibilities: incompatibilities}
+}
+
+// ======================== Hard Coded Chains ====================== //
+
+// DocumentBaseSearch is a chain that scans for incompatibilities at the base of an OpenAPI doc
+func DocumentBaseSearch(doc *openapiv3.Document) (incompatibilities []*Incompatibility) {
```

Comment:
Should each incompatibility function be in its own package?   Thoughts?

Reply:
As this expands it might make sense to separate into own files or packages but for now I'll continue under this file